src/sink/gpub.rs

```diff
@@ -129,74 +141,117 @@ impl Sink for GoogleCloudPubSub {
-                        for processed_elem in &mut processed {
-                            msg.append(processed_elem);
-                        }
-                        response.push(make_command_response(
-                            "send_message",
-                            &pubsub::send_message(remote, &project_id, &topic_name, &msg).await?,
-                        ));
-                    }
 
-                    PubSubCommand::Unknown => {
-                        warn!(
-                            "Unknown Google Cloud PubSub command: `{}` attempted",
-                            command.to_string()
-                        );
-                        return Err(format!(
-                            "Unknown Google Cloud PubSub command: `{}` attempted",
-                            command.to_string()
-                        )
-                        .into());
-                    }
-                };
-            }
+        let remote_subscriber = if let Some(remote_subscriber) = &mut self.remote_subscriber {
+            remote_subscriber
+        } else {
+            self.remote_subscriber = Some(pubsub_auth::setup_subscriber_client().await?);
+            let remote_subscriber = self
+                .remote_subscriber
+                .as_mut()
+                .ok_or("Subscriber client error!")?; //.ok_or("Subscriber client error!")?;
+            remote_subscriber
+            // TODO - Qos checks
+        };
 
-            if self.is_linked {
-                if let Some(reply_channel) = &self.reply_channel {
-                    let mut meta = Object::with_capacity(1);
-                    if let Some(correlation) = maybe_correlation {
-                        meta.insert_nocheck("correlation".into(), correlation);
+        let maybe_correlation = event.correlation_meta();
+        let mut response = Vec::new();
+        for value in event.value_iter() {
+            let command = parse_command(value)?;
+            match command {
+                PubSubCommand::SendMessage(project_id, topic_name, data) => {
+                    let mut msg: Vec<u8> = vec![];
+                    let encoded = codec.encode(&data)?;
+                    let mut processed =
+                        postprocess(&mut self.postprocessors, event.ingest_ns, encoded)?;
+                    for processed_elem in &mut processed {
+                        msg.append(processed_elem);
                     }
+                    response.push(make_command_response(
+                        "send_message",
+                        &pubsub::send_message(remote_publisher, &project_id, &topic_name, &msg)
+                            .await?,
+                    ));
+                }
+
+                PubSubCommand::CreateSubscription(
+                    project_id,
+                    topic_name,
+                    subscription_name,
+                    enable_message_ordering,
```

Comment:
this would probably be better as an boolean instead of a string, or an enum if we there are more then two values

Reply:
Yes I think these fields are a part of an enum 🤔

Reply:
Ja you define it here, you could change that along with named fields :)

https://github.com/tremor-rs/tremor-runtime/pull/1067/files/86b3ef2c2bef2fa7cefbb8726b8c7dea2c945051#diff-222dacbb009cbcb0efa06bc565c40b364ec2cd1be034a913d41fdb1672388c75R54